### DISCUSSION TITLE
Power options for testing?

### DISCUSSION THREAD

**Discussion by scytacki** [meatpiHQ/wican-fw#42] [Q&A]
In the docs there is the comment:
> The OBD2 adapter is not designed to powered off the USB connecter. The USB connector can power the adapter to flash custom firmware or hard reset the device and can also be used for debugging.

A few questions:
Can the hardware be damaged if powered via USB, or will it just cause issues when running the firmware?

Is the USB power enough to use the webserver to install firmware OTA?

If I want to test out some of the BLE stuff on the device is the usb power enough for that? 

If the usb power is not enough, do you have a simple option for giving it enough power so I can work it without sitting in the car? 
Perhaps I could use a 9V battery and clip it to the +V and GND on the OBD2 connector?

I started being concerned about this, because I was testing custom firmware using just USB power and the WiCAN went into a reboot loop. It continued to do that until I flashed the released v169 firmware. If it continues to be a problem I'll add an issue.

**Comment by meatpiHQ** [comment]
@scytacki 

> Can the hardware be damaged if powered via USB, or will it just cause issues when running the firmware?

No

> If I want to test out some of the BLE stuff on the device is the usb power enough for that?

You might get random reboots or stuck in boot loop.

> If the usb power is not enough, do you have a simple option for giving it enough power so I can work it without sitting in the car?
> Perhaps I could use a 9V battery and clip it to the +V and GND on the OBD2 connector?

Yup this would work, 7V battery is fine.

> I started being concerned about this, because I was testing custom firmware using just USB power and the WiCAN went into a reboot loop. It continued to do that until I flashed the released v169 firmware. If it continues to be a problem I'll add an issue.

The USB power can cause the boot loop. The issue here is that the power supply voltage is marginal on 5V, so you can't reliably power it from USB. I had a (never released) hardware version with different power supply that works on 5V, but it was never released because it had higher sleep current. The USB version is better for development than the OBD version.

Contact me using the link below with you preferred shipping address and I'll send you a sample of the USB version or the OBD version that works on 5V(if I can find it). This going to take a while to get to you. I suggest you use the 9V battery for now.

https://www.meatpi.com/contact
